relayer/app/cursors/watcher.go

```diff
@@ -0,0 +1,123 @@
+package cursors
+
+import (
+	"context"
+	"slices"
+
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/xchain"
+)
+
+// todo mind the query limit on db
+
+type Watcher struct {
+	cursors  CursorTable
+	provider xchain.Provider
+}
+
+func NewWatcher(cursors CursorTable, provider xchain.Provider) (*Watcher, error) {
+	return &Watcher{
+		cursors,
+		provider,
+	}, nil
+}
+
+func (w *Watcher) getStreamCursors(ctx context.Context, stream xchain.StreamID) ([]*Cursor, error) {
+	iterator, err := w.cursors.List(
+		ctx,
+		CursorPrimaryKey{}.WithSrcChainIdConfLevelDstChainId(
+			stream.SourceChainID,
+			uint32(stream.ConfLevel()),
+			stream.DestChainID,
+		))
+
+	if err != nil {
+		return nil, errors.Wrap(err, "list cursors")
+	}
+	defer iterator.Close()
+
+	var cursorsDesc []*Cursor
```

Comment:
what is `*Desc`? why not just `resp` or `cursors`?

Reply:
this is leftover from reverse ordered, keep in mind it's wip